### DISCUSSION TITLE
Password reset

### DISCUSSION THREAD

**Discussion by ricardo-cunha-zz** [community/community#34418] [Other Feature Feedback, Questions, & Ideas]
Hey guys,

I'm changing computers, and I can't remember my password, but I'm logged in to my previous computer.
I have 2FA switched on, but I can't find a way to either recover the password or get the recovery codes

I changed the password on 16.03, and I have the recovery codes from that date.
And at 16.06, I was asked to change the password again, and that's the one I, unfortunately, don’t remember.

I tried using the codes from 16.03, but they don’t work.
And I also tried to see the recover codes since I’m logged in to the account, but it requests this authentication code, but I don’t know where it’s being sent.

<img width="510" alt="image" src="https://user-images.githubusercontent.com/98830379/192706095-a5cbf021-3627-4b94-874b-b7808125f93a.png">

In conclusion, it seems that I need to know one of these two codes (recover codes or authentication code), but I don’t have any way to get them.

Any ideas to solve this?

**Comment by airtower-luna** [comment]
> And I also tried to see the recover codes since I’m logged in to the account, but it requests this authentication code, but I don’t know where it’s being sent.

Nothing gets sent anywhere. It asks for a TOTP code, which should be generated by the authenticator application you set up when enabling 2FA. You need to open the application and look at the current code for GitHub. "Current" is key: the first T in TOTP is for "time-based", if the time on the device is off by more than a couple of seconds the code won't work.

If you don't have the authenticator any more it gets really complicated. See [Recover an account with 2FA](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/recovering-your-account-if-you-lose-your-2fa-credentials) for the options you have. And definitely **do not** log out on the computer where you have access, using a verified device is one of the alternative options, so you'll want to keep that login. :warning:

**Reply by ricardo-cunha-zz** [reply]
I think I just scanned the QR code with my phone and received a code in my phone messages. 
![image](https://user-images.githubusercontent.com/98830379/192754856-1192509d-cf16-46bd-ac23-7a04ece0bcc6.png)

Regarding the recovery you suggest, I guess my only option is to request a recovery as shown in the link, but I can't see that option in my case. Do you know why?

my case
![image](https://user-images.githubusercontent.com/98830379/192755994-105f3a3e-44d4-4286-bf5f-d84273b0bc3a.png)


github documentation
![image](https://user-images.githubusercontent.com/98830379/192756113-a15c3475-de0e-4547-b213-0bd001dafd86.png)

**Reply by airtower-luna** [reply]
> I think I just scanned the QR code with my phone and received a code in my phone messages.

That is weird to say the least, for two reasons: 

* The prompt you see asks for a TOTP code, not an SMS one.
* AFAIK you shouldn't need to scan a QR code to set up SMS 2FA, while TOTP uses that as (one option) to get the shared secret into your phone.

If you're sure you've been using SMS you might want to [contact GitHub support](https://support.github.com/request) and see if they can figure out why you're not getting the SMS code and get asked about TOTP instead.

> Regarding the recovery you suggest, I guess my only option is to request a recovery as shown in the link, but I can't see that option in my case. Do you know why?

It looks like you're comparing the (already logged in) confirmation before seeing recovery codes with a login screen. Again you shouldn't log out of the working session, but you could try if you see the option when you try to log in from an "incognito mode" window or second browser instance.

**Reply by kiashok** [reply]
I am running into the same issue. @ricardo-cunha were you able to work around this problem and reset your password?
